src/components/ChainSelectionPopover/ChainSelectionPopover.tsx

```diff
@@ -0,0 +1,121 @@
+import { useRef, RefObject } from "react";
+import { useTranslation } from "react-i18next";
+
+import { ChainIds } from "@airswap/constants";
+
+import { useAppDispatch } from "../../app/hooks";
+import nativeCurrency from "../../constants/nativeCurrency";
+import { CHAIN_PARAMS } from "../../constants/supportedNetworks";
+import { setWalletConnected } from "../../features/wallet/walletSlice";
+import {
+  Container,
+  NetworksContainer,
+  NetworkButton,
+  NetworkIcon,
+} from "./ChainSelectionPopover.styles";
+import PopoverSection from "./subcomponents/PopoverSection/PopoverSection";
+
+type ChainSelectionPopoverPropsType = {
+  chainId: number | undefined;
+  account: string | undefined | null;
+  open: boolean;
+  popoverRef: RefObject<HTMLDivElement>;
+  transactionsTabOpen: boolean;
+};
+
+/**
+ * @remarks this component renders an unordered list with supported networks. Gets rendered onto ChainButton component
+ * @param chainId originates from useWeb3React in Wallet.tsx, then passed into ChainButton, then here
+ * @param account originates from useWeb3React in Wallet.tsx, then passed into ChainButton, then here
+ * @param open is a boolean value which determins whether or not to display the popover. Its value is the state value `chainsOpen` which originates in Wallet.tsx and gets passed down to ChainButton, when then gets passed into this
+ * @param transactionsTabOpen is a boolean value which indicates whether the right-side drawer that displays transactions is open. This drawer opens when a user clicks on WalletButton. This prop exists in this component so it appropriately shifts to the left when the drawer opens
+ * @returns container with a list of supported EVM networks
+ */
+const ChainSelectionPopover = ({
+  chainId,
+  account,
+  open,
+  popoverRef,
+  transactionsTabOpen,
+}: ChainSelectionPopoverPropsType) => {
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+
+  const handleNetworkSwitch = async (chainId: string) => {
```

Comment:
Handling of the network switch should be propagated back to Wallet.tsx.

Reply:
Wallet.tsx renders `ChainButton`, which renders `ChainSelectionPopover`. Wallet.tsx is a functional component that does not return any hooks, so I don't see how it can make sense to move it into Wallet, unless we want to add more clutter to the component. If Wallet.tsx was a hook, it would make more sense to me if ChainSelectionPopover was moved into it

Reply:
Just trying to gather all the `dispatch` functions so they're not lost in several smaller components. Since `dispatch` is not used anymore in this component it's fine.